contracts/Vesting.sol

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/// @title Vesting
+/// @author Eldar Dryga
+/// @notice this contract allows you to distribute tokens to users within a certain time
+
+contract Vesting is Ownable, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    IERC20 private immutable _token;
+
+    uint32 public periodOfDistributionInDays;
+    uint32 public amountOfDistributionPerUser;
+    uint128 public users;
+    uint256 public maxUsers;
+
+    struct UsersInfo {
+        uint256 userTime;
+        uint256 claimedRevard;
+        uint256 allClaimedRevard;
+        uint256 finalDistributionTime;
+    }
+    mapping(address => UsersInfo) public infoOfUser;
+
+    constructor(address token_) {
+        require(token_ != address(0));
```

Comment:
in require statement add error message, so if u pass zero address on deploy it will throw readable error but nor just reverted